x-pack/elastic-agent/pkg/agent/cmd/enroll.go

```diff
@@ -116,6 +118,19 @@ func enroll(streams *cli.IOStreams, cmd *cobra.Command, flags *globalFlags, args
 	}
 
 	err = c.Execute()
+	signal := make(chan struct{})
+
+	backExp := backoff.NewExpBackoff(signal, 30*time.Second, 10*time.Minute)
+
+	for err == fleetapi.ErrTooManyRequests {
+		fmt.Fprintln(streams.Out, "Too many requests on the remote server, will retry in a moment.")
```

Comment:
Is it possible to print the wait time here? With 10min max someone might get impatient and think the beat is deadlocked